com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -293,7 +293,10 @@ public void PutObservations(string behaviorName, AgentInfo info, List<ISensor> s
             {
                 m_OrderedAgentsRequestingDecisions[behaviorName] = new List<int>();
             }
-            m_OrderedAgentsRequestingDecisions[behaviorName].Add(info.episodeId);
+            if (!info.done)
+            {
+                m_OrderedAgentsRequestingDecisions[behaviorName].Add(info.episodeId);
+            }
             if (!m_LastActionsReceived.ContainsKey(behaviorName))
```

Comment:
This is to make sure that the agents that are done will not be expecting a decision.